OOP_HW1/src/test/java/ut_ticketsystem.java

```diff
@@ -0,0 +1,227 @@
+import org.junit.jupiter.api.Test;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.catchThrowable;
+
+import java.util.Vector;
+
+public class ut_ticketsystem {
+    @Test
+    public void test_check_ticket_price_should_have_correct_price() {
+
+        Order ticket = new Order(TicketType.ADULT, 1);
+        int price = TicketSystem.CheckTicketPrice(ticket);
+
+        assertThat(price).matches(q -> q == 350);
+    }
+
+    @Test
+    public void test_check_ticket_price_should_throw_exception() {
+
+        Order ticket = new Order();
+        ticket.setTicketType(TicketType.ADULT);
+        ticket.setQuantity(-1);
+
+        Throwable thrown = catchThrowable(() -> {
+            TicketSystem.CheckTicketPrice(ticket);
+        });
+
+        assertThat(thrown)
+                .isInstanceOf(IllegalArgumentException.class)
+                .hasMessageContaining("There is at least one ticket.");
+
+    }
+
+    @Test
+    public void test_check_ticket_price_should_give_a_discount() {
+
+        Order ticket = new Order();
+        ticket.setTicketType(TicketType.ELDER);
+        ticket.setQuantity(10);
+
+        int price = TicketSystem.CheckTicketPrice(ticket);
+
+        assertThat(price).matches(q -> q == 1920);
+
+    }
+
+    @Test
+    public void test_check_today_sales_have_correct_price() {
+
+        Order adult_ticket = new Order(), child_ticket = new Order(), elder_ticket = new Order();
+        adult_ticket.setQuantity(10);
+        adult_ticket.setTicketType(TicketType.ADULT);
+
+        child_ticket.setQuantity(5);
+        child_ticket.setTicketType(TicketType.CHILD);
+
+        elder_ticket.setQuantity(3);
+        elder_ticket.setTicketType(TicketType.ELDER);
```

Comment:
Since you have constructor to construct the Order class, why not just use constructor to construct your object?

```suggestion
        Order adult_ticket = new Order(10, TicketType.ADULT), child_ticket = new Order(5, TicketType.CHILD), elder_ticket = new Order(3, TicketType.ELDER);
```